src/contract_wrappers/modules/checkpoint/erc20_dividend_checkpoint_wrapper.ts

```diff
@@ -0,0 +1,578 @@
+        excluded: string[],
+    }
+    
+    interface IPushDividendPaymentParams extends ITxParams {
+        dividendIndex: BigNumber,
+        start: BigNumber,
+        iterations: BigNumber,
+    }
+    
+    interface IGetDividendProgressParams {
+        dividendIndex: BigNumber
+    }
+    
+    interface ICreateDividendParams extends ITxParams {
+        maturity: BigNumber,
+        expiry: BigNumber,
+        token: string,
+        amount: BigNumber,
+        name: string,
+    }
+    
+    interface ICreateDividendWithCheckpointParams extends ITxParams {
+        maturity: BigNumber,
+        expiry: BigNumber,
+        token: string,
+        amount: BigNumber,
+        checkpointId: BigNumber,
+        name: string,
+    }
+    
+    interface ICreateDividendWithExclusionsParams extends ITxParams {
+        maturity: BigNumber,
+        expiry: BigNumber,
+        token: string,
+        amount: BigNumber,
+        excluded: string[],
+        name: string,
+    }
+
+    interface ICreateDividendWithCheckpointAndExclusionsParams extends ITxParams {
+        maturity: BigNumber,
+        expiry: BigNumber,
+        token: string,
+        amount: BigNumber,
+        checkpointId: BigNumber,
+        excluded: string[],
+        name: string,
+    }
+    
+    interface IReclaimDividendParams extends ITxParams {
+        dividendIndex: BigNumber
+    }
+    
+    interface IWithdrawWithholdingParams extends ITxParams {
+        dividendIndex: BigNumber
+    }
+  
+  /**
+   * This class includes the functionality related to interacting with the ERC20DividendCheckpoint contract.
+   */
+  export class ERC20DividendCheckpointWrapper extends ContractWrapper {
+    public abi: ContractAbi = ERC20DividendCheckpoint.abi;
+    private polymathRegistry: PolymathRegistryWrapper;
+    private erc20DividendCheckpointContract: Promise<ERC20DividendCheckpointContract>;
+  
+    /**
+     * Instantiate ERC20DividendCheckpointWrapper
+     * @param web3Wrapper Web3Wrapper instance to use
+     * @param polymathRegistry The PolymathRegistryWrapper instance contract
+     */
+    constructor(web3Wrapper: Web3Wrapper, polymathRegistry: PolymathRegistryWrapper) {
+      super(web3Wrapper);
+      this.polymathRegistry = polymathRegistry;
+      this.erc20DividendCheckpointContract = this._getERC20DividendCheckpointContract();
+    }
+  
+    /**
+     * Returns the contract address
+     */
+    public getAddress = async (): Promise<string> => {
+      return (await this.erc20DividendCheckpointContract).address;
+    }
+  
+    public setWithholdingFixed = async (params: ISetWithholdingFixedParams) => {
+        return async () => {
+            return (await this.erc20DividendCheckpointContract).setWithholdingFixed.sendTransactionAsync(
+                params.investors,
+                params.withholding,
+            );
+        }
+    }
+
+    public EXCLUDED_ADDRESS_LIMIT = async (): Promise<BigNumber> => {
```

Comment:
This can be removed. Every method in uppercase are constants in the contracts. I think we don't need them for now.